test/remote/gateways/remote_pay_trace_test.rb

```diff
@@ -0,0 +1,201 @@
+    assert_equal 'One or more errors has occurred.', response.message
+  end
+
+  def test_successful_refund
+    purchase = @gateway.purchase(200, @credit_card, @options)
+    assert_success purchase
+    authorization = purchase.authorization
+
+    settle = @gateway.settle()
+    assert_success settle
+
+    refund = @gateway.refund(authorization, @options.merge(amount: 1.00))
+    assert_success refund
+    assert_equal 'Your transaction was successfully refunded.', refund.message
+  end
+
+  def test_partial_refund
+    purchase = @gateway.purchase(400, @mastercard, @options)
+    assert_success purchase
+    authorization = purchase.authorization
+
+    settle = @gateway.settle()
+    assert_success settle
+
+    refund = @gateway.refund(authorization, @options.merge(amount: 3.00))
+    assert_success refund
+    assert_equal 'Your transaction was successfully refunded.', refund.message
+  end
+
+  def test_refund_without_amount
+    purchase = @gateway.purchase(@amount, @discover, @options)
+    assert_success purchase
+    authorization = purchase.authorization
+
+    settle = @gateway.settle()
+    assert_success settle
+
+    refund = @gateway.refund(authorization)
+    assert_success refund
+    assert_equal 'Your transaction was successfully refunded.', refund.message
+  end
+
+  def test_failed_refund
+    response = @gateway.refund('', @options)
+    assert_failure response
+    assert_equal 'One or more errors has occurred.', response.message
+  end
+
+  def test_successful_void
+    auth = @gateway.authorize(@amount, @amex, @options)
+    assert_success auth
+
+    assert void = @gateway.void(auth.authorization)
+    assert_success void
+    assert_equal 'Your transaction was successfully voided.', void.message
+  end
+
+  def test_failed_void
+    response = @gateway.void('')
+    assert_failure response
+    assert_equal 'One or more errors has occurred.', response.message
+  end
+
+  def test_successful_verify
+    response = @gateway.verify(@mastercard, @options)
+    assert_success response
+    assert_equal 'Your transaction was successfully approved.', response.message
+  end
+
+  def test_successful_store_and_redact_customer_profile
+    response = @gateway.store(@mastercard, @options)
+    assert_success response
+    customer_id = response.params['customer_id']
+    redact = @gateway.redact(customer_id)
+    assert_success redact
+    assert_equal true, redact.success?
+  end
+
+  def test_duplicate_customer_creation
+    create = @gateway.store(@discover, @options)
+    customer_id = create.params['customer_id']
+    response = @gateway.store(@discover, @options.merge(customer_id: customer_id))
+    assert_failure response
+    assert_equal false, response.success?
+  end
+
+  # Not including a test_failed_verify since the only way to force a failure on this gateway is with a specific dollar amount. Since verify is auth and void combined, having separate tests for auth and void should suffice.
```

Comment:
i agree! (also it would be neat to break this comment up into a few shorter lines for the benefit of those of us who insist on using ancient editors)

Reply:
Will do! 😅